equinox/nn/_embedding.py

```diff
@@ -194,9 +197,14 @@ def precompute_freqs_cis(
         freqs = 1.0 / (
             theta ** (jnp.arange(0, embedding_size, 2)[jnp.newaxis, :] / embedding_size)
         )
-        t = jnp.arange(end)
+
+        t = jnp.arange(end, dtype=float)
         freqs_outer = jnp.outer(t, freqs)
-        freqs_cis = jnp.cos(freqs_outer) + jnp.sin(freqs_outer) * 1j
+        freqs_cis = (
+            jnp.array(jnp.cos(freqs_outer), dtype=jnp.complex64)
+            + jnp.array(jnp.sin(freqs_outer), dtype=jnp.complex64) * 1j
```

Comment:
Likewise I'm not wild about the explicit dtype casts here, which may do the wrong thing if e.g. float64s (and thus complex128s) are what is being used for some reason. Or probably more likely, if e.g. bfloat16s are being used... are we going to the correct thing in all these cases?

If you're just trying to have the strict-dtype-promotion check not yell at you, then the easiest way to do it is to explicitly disable it locally where you know that what you really want the dtype promotion:
```python
with jax.numpy_dtype_promotion("standard"):
    ...
```